### DISCUSSION TITLE
Explicit valence for atom # 2 N, 5, is greater than permitted

### DISCUSSION THREAD

**Discussion by amir-tagh** [rdkit/rdkit#4742] [Q&A]
Hello,

I am using qed and rdkit to calculate the QED value for a set of SMILES. It seems there is a problem with some of the SMILES and calculations stop when encountering the problematic SMILES. Is there any way to continue the calculations and ignore the problematic SMILES?

Here is the code:

from qed import qed
from rdkit import Chem
import sys

for line in lines_1:
mol = Chem.MolFromSmiles(line)
#RDLogger.DisableLog('rdApp.*')
p = qed.properties(mol)

error message: Explicit valence for atom # 2 N, 5, is greater than permitted

I have 500k SMILES and calculations stop at 93395.

Thanks in forward.
Best,
Amir

**Comment by greglandrum** [comment]
@amir-tagh, there are *a lot* of examples out there which show how to handle this. A very quick web search would have turned up useful results.

The short answer: `MolFromSmiles()` returns `None` when there are chemistry problems. You just need to detect that.

**Comment by amir-tagh** [comment]
Hi Greg,

Thanks for the hint, solved.

Cheers,
